### PR TITLE
Instrument MME state machine

### PR DESCRIPTION
**Pull request by bnordbo**
Besides structuring the Prometheus client code a bit, this adds
counters to the various message types of the MME state machine.
The implementation is a bit clunky as Open5GS is built with C89.

### CODE REVIEW

File: src/mme/mme-metrics.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    "Open5GS MME number of eNB",
+                    0,
+                    NULL));
+    mme_sessions_gauge = prom_collector_registry_must_register_metric(
+            prom_gauge_new(
+                    "open5gs_mme_session_gauge",
+                    "Open5GS MME number of sessions",
+                    0,
+                    NULL));
+    const char* messagesCounterLabels[5] = {"s1ap", "emm", "esm", "s6a", "s11"};
```
Comment (nicholas-wg2) [inline src/mme/mme-metrics.c]: what if we define these constants outside of the function (ie. global scope of mme-metrics.c) then we wouldn't need to define them again in each function they are used within the code?

File: src/mme/mme-sm.c [hunk 2]
```diff
@@ -239,6 +241,8 @@ void mme_state_operational(ogs_fsm_t *s, mme_event_t *e)
         if (rc == OGS_OK) {
             e->enb = enb;
             e->s1ap_message = &s1ap_message;
+            const char* s1apLabel[1] = {"s1ap"};
```
Comment (nicholas-wg2) [inline src/mme/mme-sm.c]: i'll need to try it out, but wouldn't it make sense to not use an array here and just define it as:
`const char* s1apLabel = "s1ap"`
really not sure here, but this relates to previous comment about declaring the list of labels ahead of time in mme-metrics.c and just re-using them (or maybe we just use pre-processor defines?)

Comment (bnordbo) [reply src/mme/mme-sm.c]: Good idea! That solves to problems in one go, and it seems to work. Err.. build, at least. :-)

**Comment by bnordbo** [comment]
I was hoping to make the tests pass, but alas only the first four works. But the same is true for the latest upstream commit too, so ... well. @nicholas-wg2, have you had any luck with the tests? I get:  

```
Found ninja-1.10.0 at /usr/bin/ninja
[105/106] Running all tests.
1/9 open5gs:unit / core                     OK       7.74 s
2/9 open5gs:unit / crypt                    OK       0.13 s
3/9 open5gs:system / sctp                   OK       0.05 s
4/9 open5gs:unit / unit                     OK       0.13 s
5/9 open5gs:epc / simple                    FAIL     1.08 s (killed by signal 11 SIGSEGV)
6/9 open5gs:epc / mnc3                      FAIL     1.91 s (killed by signal 11 SIGSEGV)
7/9 open5gs:epc / volte                     FAIL     2.96 s (killed by signal 6 SIGABRT)
8/9 open5gs:epc / csfb                      FAIL     2.84 s (killed by signal 11 SIGSEGV)
9/9 open5gs:5gc / cups                      FAIL     2.96 s (killed by signal 6 SIGABRT)
```

**Comment by nicholas-wg2** [comment]
could you find the error that is actually causing the failure?
can go into the `build` folder and run `meson test -v`

i was getting these errors:

```
socket-test         : -09/08 09:31:03.321: [sock] ERROR: socket bind(10) [::]:7777 failed (98:Address already in use) (../lib/core/ogs-socket.c:109)
09/08 09:31:03.321: [sock] ERROR: tcp_server() [::]:7777 failed (98:Address already in use) (../lib/core/ogs-tcp.c:55)
|Line 101: expected non-NULL, but saw NULL
\09/08 09:31:03.321: [sock] FATAL: ogs_sock_accept: Assertion `sock' failed. (../lib/core/ogs-socket.c:169)
```

and

```
Failed to bind to port 0: Address already in use
09/08 09:31:05.015: [sbi] ERROR: Cannot start SBI server (../lib/sbi/server.c:279)
Failed to bind to port 0: Address already in use
09/08 09:31:05.015: [sbi] ERROR: Cannot start SBI server (../lib/sbi/server.c:279)
```

had to make sure tunnel was up as mentioned in start of https://open5gs.org/open5gs/docs/guide/02-building-open5gs-from-sources/ (can also run the script `./misc/netconf.sh`)

also sometimes the application was already running so had to kill them to free up the port

```
netstat -anp | grep open5gs


sudo pkill -9 open5gs-hssd
sudo pkill -9 open5gs-pcrfd
sudo pkill -9 open5gs-pgwd
sudo pkill -9 open5gs-sgwd
sudo pkill -9 open5gs-mmed
sudo pkill -9 open5gs-nrfd
```

finally, the author was suggesting to make changes to sleep timer during tests (from approx 1s to 5s) to wait for freeDiameter to come up and all services listening
mentioned in an issue 549 of the open5gs repo
although i noticed we don't have any sleep timers in our current src, so had to pull updates from upstream and now i'm not getting any more build errors!

**Comment by bnordbo** [comment]
I still only get the first four tests passing:

```
bn@epc-dev:~/build$ ninja -C build test
ninja: Entering directory `build'
[0/1] Running all tests.
 1/10 open5gs:unit / core                     OK       7.72 s
 2/10 open5gs:unit / crypt                    OK       0.09 s
 3/10 open5gs:app / sctp                      OK       0.07 s
 4/10 open5gs:unit / unit                     OK       0.12 s
 5/10 open5gs:5gc / registration              TIMEOUT 30.14 s
 6/10 open5gs:epc / attach                    TIMEOUT 30.13 s
 7/10 open5gs:epc / volte                     TIMEOUT 30.12 s
 8/10 open5gs:epc / csfb                      TIMEOUT 30.06 s
 9/10 open5gs:app / 310014                    TIMEOUT 30.09 s
10/10 open5gs:app / handover                  TIMEOUT 30.07 s
```

My guess is that something's failing to start, but it's not very helpful. Anyway, if you can build it, it should be enough and we can probably make an ad-hoc process for deploying the latest from HEAD?

**Comment by nicholas-wg2** [comment]
i also found a lot of the test failures i was getting was because the httpd was failing to start, which is called from app-init.c for both pcrf and mme which mistakenly had same port defined (but no error message to indicate why the app failed to start

**Pull request merged**